prow/githubeventserver/githubeventserver.go

```diff
@@ -92,9 +93,35 @@ func (g *GitHubEventServer) GetHTTPServeMux() *http.ServeMux {
 
 // Listen runs an http server by using the existing http.ServeMux, port and the
 // given grace period duration
-func (g *GitHubEventServer) Listen(gracePeriod time.Duration) {
+func (g *GitHubEventServer) Listen(ctx context.Context, gracePeriod time.Duration) {
 	defer interrupts.WaitForGracefulShutdown()
-	interrupts.ListenAndServe(&http.Server{Addr: ":" + strconv.Itoa(g.port), Handler: g.httpServeMux}, gracePeriod)
+
+	server := &http.Server{Addr: ":" + strconv.Itoa(g.port), Handler: g.httpServeMux}
+	interrupts.Run(func(ctx context.Context) {
```

Comment:
The caller uses `interrupts.Run()`. The server here just runs. If you are literally just running an HTTP server, expose it and allow `interrupts.ListenAndServe()`.